proxyserver/middleware/versioned_writes.go

```diff
@@ -334,7 +334,7 @@ func (v *versionedWrites) headObject(writer http.ResponseWriter, req *http.Reque
 
 func (v *versionedWrites) handleObjectDeleteStack(writer http.ResponseWriter, request *http.Request, account, container, versionsContainer, object string) {
 	ctx := GetProxyContext(request)
-	listingPath := fmt.Sprintf("/v1/%s/%s?format=json&prefix=%s&reverse=on", account, versionsContainer, v.versionedObjectPrefix(object))
+	listingPath := fmt.Sprintf("/v1/%s/%s?format=json&prefix=%s&reverse=on", common.Urlencode(account), common.Urlencode(versionsContainer), url.QueryEscape(v.versionedObjectPrefix(object)))
```

Comment:
Does this + containerListing() double encode?

Reply:
Oh that one doesn't? I can't read.

Reply:
:)

containerListing does not encode. That's why I changed its parameter name to urlStr. But yeah, all that diffy stuff gets confusing. :)